base/applications/cmdutils/where/strlist.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * PROJECT:     ReactOS WHERE command
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Providing string list
+ * COPYRIGHT:   Copyright 2021 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+#pragma once
+
+static LPWSTR str_clone(LPCWSTR psz)
+{
+    size_t cb;
+    LPWSTR pszNew;
+    if (!psz)
+        return NULL;
+    cb = (wcslen(psz) + 1) * sizeof(WCHAR);
+    pszNew = (LPWSTR)malloc(cb);
+    if (!pszNew)
+        return NULL;
+    memcpy(pszNew, psz, cb);
+    return pszNew;
+}
+
+typedef struct strlist_t
+{
+    LPWSTR *ppsz;
+    int count;
+} strlist_t;
+#define strlist_default { NULL, 0 }
+
+static inline void strlist_init(strlist_t *plist)
+{
+    plist->ppsz = NULL;
+    plist->count = 0;
+}
+
+static inline LPWSTR strlist_get_at(strlist_t *plist, int i)
+{
+    return plist->ppsz[i];
+}
+
+static int strlist_add(strlist_t *plist, LPWSTR psz)
```

Comment:
Since all your strlist_add() calls seem to systematically duplicate the string before, you may as well just call the duplicate within that function. That would allow to make this strlist module self-contained, and would also justify that on failures, this function can indeed call `free(psz);`, and the strlist_destroy() function call `free(plist->ppsz[i]);` for each string pointer.

Reply:
OK.